Neo4j.Driver/Neo4j.Driver/Internal/Protocol/Utility/FakeTransaction.cs

```diff
@@ -0,0 +1,90 @@
+﻿// Copyright (c) "Neo4j"
+// Neo4j Sweden AB [http://neo4j.com]
+// 
+// This file is part of Neo4j.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License").
+// You may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Neo4j.Driver.Internal;
+
+/// <summary>
+/// Supports cursors on auto commit functions so we don't need to null check and check error.
+/// this will always return false for the transaction has errored because the transaction doen't exist.
+/// </summary>
+internal sealed class FakeTransaction : IInternalAsyncTransaction
```

Comment:
Could there be a better name for this class? Fake sounds a bit like it's something to do with testing.